tests/test_basic.py

```diff
@@ -254,6 +256,57 @@ def test_signature_overwriting(self):
             golem_messages.dump(msg, self.ecc.raw_privkey, self.ecc.raw_pubkey)
 
 
+class VerifyMessageSignatureTest(unittest.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.keys = cryptography.ECCx(None)
+        self.keys2 = cryptography.ECCx(None)
+
+    def add_sig(self, msg):
+        golem_messages.dump(msg, self.keys.raw_privkey, None)
+        self.assertIsNotNone(msg.sig)
+
+    def test_verify(self):
+        msg = message.Hello()
+        self.add_sig(msg)
+        msg.verify_signature(self.keys.raw_pubkey)
+
+    def test_verify_nosig(self):
+        msg = message.Hello()
+        self.assertIsNone(msg.sig)
+        with self.assertRaises(exceptions.CoincurveError):
```

Comment:
This probably should also raise `InvalidSignature` to be consistent
` raise InvalidSignature from parent_exception`